### PR TITLE
support the bestEffort flag

### PR DESCRIPTION
**Pull request by slackhappy**
Support `CLD2::kCLDFlagBestEffort` for shorter text

Helps with issues like #78 for example:

without bestEffort
```
> console.log((await cld.detect("J'aime me beurrer la biscote")))
Uncaught Error: Failed to identify language
    at Object.detect (/Users/jgallagher/extsrc/slackhappy.gh/node-cld/index.js:76:15)
    at async REPL6:1:46
```

with bestEffort:

```
> console.log((await cld.detect("J'aime me beurrer la biscote", {bestEffort:true})))
{
  reliable: true,
  textBytes: 30,
  languages: [ { name: 'FRENCH', code: 'fr', percent: 96, score: 176 } ],
  chunks: []
}
```

Or #31 

```
> console.log((await cld.detect("beija-flor: laíla defende maior número de parcerias e critica altos gastos")))
Uncaught Error: Failed to identify language
    at Object.detect (/Users/jgallagher/extsrc/slackhappy.gh/node-cld/index.js:76:15)
    at async REPL9:1:46
```

with bestEffort:

```
> console.log((await cld.detect("beija-flor: laíla defende maior número de parcerias e critica altos gastos", {bestEffort:true})))
{
  reliable: true,
  textBytes: 77,
  languages: [ { name: 'PORTUGUESE', code: 'pt', percent: 98, score: 215 } ],
  chunks: []
}
```

#33 

```
> console.log(await cld.detect('Hi there'))
Uncaught Error: Failed to identify language
    at Object.detect (/Users/jgallagher/extsrc/slackhappy.gh/node-cld/index.js:76:15)
    at async REPL3:1:45
> 
```

with bestEffort:

```
> console.log(await cld.detect('Hi there', {bestEffort:true}))
{
  reliable: true,
  textBytes: 10,
  languages: [ { name: 'ENGLISH', code: 'en', percent: 90, score: 796 } ],
  chunks: []
}
```

### CODE REVIEW

**Review by dachev — APPROVED** [review]

**Comment by dachev** [comment]
@slackhappy thank you!

**Comment by slackhappy** [comment]
oops sorry - one more update to the typescript defs and the docs

File: README.md [hunk 1]
```diff
@@ -91,6 +91,11 @@ Pass top level domain as a hint
 
 Pass an HTTP "Content-Encoding" value as a hint
 
+#### bestEffort
+
+Set to true to give best-effort answer, instead of UNKNOWN_LANGUAGE. May be useful for
```
Comment (slackhappy) [inline README.md:96]: description is from https://github.com/dachev/node-cld/blob/1a135708e99f1d5e10ecb70197b2b42fe42d1006/deps/cld/public/compact_lang_det.h#L373

**Comment by slackhappy** [comment]
@dachev, ready for re-review, I added documentation and updated the ts defs. Thanks so much for considering this PR!

**Comment by dachev** [comment]
@slackhappy thanks for your contribution. I pushed the changes to npm: https://www.npmjs.com/package/cld?activeTab=versions

**Comment by slackhappy** [comment]
@dachev, thank you so much!

**Pull request merged**